### PR TITLE
Created role schema and permissions

### LINKED ISSUE #54: User Management and Roles
- create the author and reviewer and editor roles, with respective permissions

### PR DESCRIPTION
**Pull request by KS963000**
fixes #54

### CODE REVIEW

**Comment by KS963000** [comment]
I have made the final changes to #54, could you please review the PR?

**Comment by ravinder-chadha** [comment]
Admin can manage the permissions of others as well, please add that role too.

**Comment by KS963000** [comment]
Done, please check once.

File: backend/controllers/roleController.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+const Role = require('../models/roleModel');
+
+class RoleController {
+    async createRoles() {
```
Comment (ravinder-chadha) [inline backend/controllers/roleController.js:4]: There is only createRoles, but we might need to deleteRole? modifyRole? right

File: backend/controllers/roleController.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            permissions: ['READ', 'UPDATE']
+        });
+
+        const coAuthorRole = new Role({
+            name: 'co-author',
+            permissions: ['READ','CONTRIBUTE']
+        });
+
+        const adminRole = new Role({
+            name: 'author',
```
Comment (ravinder-chadha) [inline backend/controllers/roleController.js]: the adminRole has the same name as author

**Review by ravinder-chadha — COMMENTED** [review]
Also you might want to check if the roles already exist before trying to create them to avoid duplicates.

**Comment by KS963000** [comment]
Done, please check once.

File: backend/controllers/roleController.js (hunk 2 above)
Comment (ravinder-chadha) [reply backend/controllers/roleController.js]: Hi @KS963000 ,
Please rectify this asap, so that we can merge this.

**Comment by ravinder-chadha** [comment]
Hi @KS963000 ,
Please resolve all the above comments

**Comment by KS963000** [comment]
Resolved

**Review by ravinder-chadha — APPROVED** [review]

**Pull request merged**